### ISSUE TITLE
`Layout/FirstArgumentIndentation` suggests non-ideal indentation for multi-line method calls that disregards vertical alignment of Hashes

### ISSUE THREAD

**Issue by postmodern** [rubocop/rubocop#11566]
I noticed that the `Layout/FirstArgumentIndentation` gives incorrect recommendations for when calling a method who's arguments are very long and thus must be placed on the following line:

```ruby
hash = {
  'foo' => foo(
            'really long input that needs to be on its own line'
           ),
  'bar' => 'bar'
}
```

 rubocop suggests "Indent the first argument one step more than the start of the previous line." However, that would look like this, which is not ideal syntax as it breaks the vertical alignment of the Hash values:
 
```ruby
hash = {
  'foo' => foo(
    'really long input that needs to be on its own line'
  ),
  'bar' => 'bar'
}
```

### `Gemfile`

```ruby
source 'https://rubygems.org'

gem 'rubocop'
```

### `test.rb`

```ruby
def foo(arg)
  "foo #{arg}"
end

hash = {
  'foo' => foo(
            'really long input that needs to be on its own line'
           ),
  'bar' => 'bar'
}

puts hash.inspect
```

## Expected behavior

No warnings.

## Actual behavior

```
For /home/postmodern/test/ruby/rubocop: Default configuration from /home/postmodern/test/ruby/rubocop/vendor/bundle/ruby/3.1.0/gems/rubocop-1.45.1/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /home/postmodern/.cache/rubocop_cache/d6b28bd02c9e5f8ec174c433a9aa2c5d97f14ca4/6d7a3b621ca1730e04accd938619e4bdab66cfb1/f72bbb02b04a002dc6bce346080d42a545bccb45
Loading cache from /home/postmodern/.cache/rubocop_cache/d6b28bd02c9e5f8ec174c433a9aa2c5d97f14ca4/6d7a3b621ca1730e04accd938619e4bdab66cfb1/8fd162d35d0095448168aafe1f2486a827ca3df9
Inspecting 2 files
Scanning /home/postmodern/test/ruby/rubocop/Gemfile
Loading cache from /home/postmodern/.cache/rubocop_cache/d6b28bd02c9e5f8ec174c433a9aa2c5d97f14ca4/6d7a3b621ca1730e04accd938619e4bdab66cfb1/f72bbb02b04a002dc6bce346080d42a545bccb45
CScanning /home/postmodern/test/ruby/rubocop/test.rb
Loading cache from /home/postmodern/.cache/rubocop_cache/d6b28bd02c9e5f8ec174c433a9aa2c5d97f14ca4/6d7a3b621ca1730e04accd938619e4bdab66cfb1/8fd162d35d0095448168aafe1f2486a827ca3df9
C

Offenses:

Gemfile:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
source 'https://rubygems.org'
^
test.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
def foo(arg)
^
test.rb:7:13: C: [Correctable] Layout/FirstArgumentIndentation: Indent the first argument one step more than the start of the previous line.
            'really long input that needs to be on its own line'
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test.rb:8:12: C: [Correctable] Layout/ClosingParenthesisIndentation: Indent ) to column 10 (not 11)
           ),
           ^

2 files inspected, 4 offenses detected, 4 offenses autocorrectable
Finished in 0.2867879120003636 seconds
```

## Steps to reproduce the problem

1. `bundle install`
2. `bundle exec rubocop`

## RuboCop version

```
$ bundle exec rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.24.1, running on ruby 3.1.3) [x86_64-linux]
```

**Comment by koic** [comment]
Can you set an alternative configuration`EnforcedStyle: consistent_relative_to_receiver` in your .rubocop.yml? That would solve it.

```yaml
Layout/FirstArgumentIndentation:
  EnforcedStyle: consistent_relative_to_receiver
```

https://docs.rubocop.org/rubocop/1.45/cops_layout.html#layoutfirstargumentindentation

**Comment by postmodern** [comment]
@koic that fixes the example when the "first argument" of a method call is within a Hash. However, it causes warnings for regular method calls:

```yaml
AllCops:
  NewCops: enable
  SuggestExtensions: false
  TargetRubyVersion: 3.1

Layout/FirstArgumentIndentation:
  EnforcedStyle: consistent_relative_to_receiver
```

```ruby
def really_long_method_name_that_needs_its_own_line
  'foo'
end

def reverse(str)
  str.reverse
end

value = reverse(
  really_long_method_name_that_needs_its_own_line
)
```

```
test.rb:10:3: C: [Correctable] Layout/FirstArgumentIndentation: Indent the first argument one step more than reverse(.
  really_long_method_name_that_needs_its_own_line
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
